### PR TITLE
Added force sync param to party and tavern requests

### LINKED ISSUE #7407: Chat Rolls Back in Party and Tavern(?) -- must click Fetch Recent Messages to see posts and clear notification
My messages in the Tavern were up-to-date, but after I went to a Guild and then returned to the Tavern, I  found that the most recently displayed messages were gone. I had to click Fetch Recent Messages to regain them.

Possibly related, noted by Alys: "When you go to the party page, the messages no longer load automatically, and the new-message notification in the guild/party notification icon won't clear automatically. The workaround is to click the "fetch recent messages" button. "  She also noted, "Probably same issue as Tavern because both don't use Groups IDs."

_edit by Alys;_

This also prevents you seeing your own message immediately after you post it, making it look as if your message has not been posted (https://github.com/HabitRPG/habitrpg/issues/7575).
### What to do when this happens to you:

**As soon as you visit your party or Tavern page, click the Fetch Recent Messages button. That will show you all messages and clear the notification icon.**

**When you post a message, also click that button and then you'll see that your message has been saved to the party chat.**

**You do not need to comment on this issue because we know it happens for all players.**

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes #7407
### Changes

I have forced the sync every time you are on the party/tavern page. This should help update chat and prevent the need for reloads.

---

UUID:

### CODE REVIEW

**Comment by TheHollidayInn** [comment]
`Cannot find module 'to-iso-string'` npm install failed. I'm not sure how to update that.

**Comment by crookedneighbor** [comment]
I have a few adjustments I'm making.

**Comment by TheHollidayInn** [comment]
To the API-v3 branch?

**Comment by crookedneighbor** [comment]
To this PR.

**Comment by crookedneighbor** [comment]
Closing this. Re-open it against the `develop` branch.

File: website/client/js/controllers/tavernCtrl.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 habitrpg.controller("TavernCtrl", ['$scope', 'Groups', 'User', 'Challenges',
     function($scope, Groups, User, Challenges) {
-      Groups.tavern()
+      Groups.tavern(true)
```
Comment (crookedneighbor) [inline website/client/js/controllers/tavernCtrl.js:5]: We shouldn't force a lookup of the Tavern every time, that'd be a _lot_ of requests. 

I think the source of the chat bug is that the cached version of the Tavern isn't persisting the new messages that get added.

**Comment by TheHollidayInn** [comment]
@crookedneighbor Are you able to replicate the tavern part of this bug locally?

**Pull request closed without merging**